File: python/src/exercises/bob/bob.py
def is_all_caps(text: str):
    return text.isupper()


def yell_question(text: str):
    return text.isupper() and text[-1] == "?"


def is_silence(text: str):
    return len(text.strip()) == 0


def response(hey_bob):
    if yell_question(hey_bob):
        return "Calm down, I know what I'm doing!"
    elif is_silence(hey_bob):
        return "Fine. Be that way!"
    elif hey_bob.strip()[-1] == "?":
        return "Sure."
    elif is_all_caps(hey_bob):
        return "Whoa, chill out!"
    else:
        return "Whatever."


# print(response(""))

File: java/src/main/java/com/ybsid/exercism/medium/Bob.java
package com.ybsid.exercism.medium;

/**
 * <a href="https://exercism.org/tracks/java/exercises/bob">...</a>
 */
class Bob{
    public String hey(String s){
        s = s.strip();
        if (s.isEmpty()) return "Fine. Be that way!";
        boolean isAllCaps = s.matches(".*[A-Z].*") && s.equals(s.toUpperCase());
        boolean isQuestion = s.endsWith("?");
        if (isAllCaps && isQuestion) return "Calm down, I know what I'm doing!";
        if (isAllCaps)               return "Whoa, chill out!";
        if (isQuestion)              return "Sure.";
        return "Whatever.";
    }
}